src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -269,77 +262,71 @@ export class StudyViewPageStore {
             let patientAttributeCount = 0;
             let filterAttributes: ClinicalAttribute[] = []
             queriedAttributes.forEach(attribute => {
-                if(attribute.patientAttribute){
-                    if(patientAttributeCount<10){
+                if (attribute.patientAttribute) {
+                    if (patientAttributeCount < 10) {
                         filterAttributes.push(attribute)
                         patientAttributeCount++;
                     }
-                } else{
-                    if(sampleAttributeCount<10){
+                } else {
+                    if (sampleAttributeCount < 10) {
                         filterAttributes.push(attribute)
                         sampleAttributeCount++;
                     }
                 }
             });
             return filterAttributes;
         },
-        default: [],
-        onResult:(attributes:ClinicalAttribute[])=>{
-            attributes.forEach(attribute=>{
-                this._attributeChartVisibility.set(attribute.clinicalAttributeId,true);
-            })
-        }
+        default: []
     });
 
-    readonly initalClinicalDataCounts = remoteData<{[id:string]:ClinicalDataCount[]}>({
+    readonly initalClinicalDataCounts = remoteData<{ [id: string]: ClinicalDataCount[] }>({
         await: () => {
-            return this.defaultVisibleAttributes.result.map(attribute => {
-                return this.studyViewClinicalDataCountsCache.get({ attribute: attribute, filters: {} as any })
+            let promises = this.defaultVisibleAttributes.result.map(attribute => {
+                return this.studyViewClinicalDataCountsCache.get({ attribute: attribute, filters: this.emptyFilter })
             })
+            return [this.defaultVisibleAttributes, ...promises]
         },
         invoke: async () => {
             return _.reduce(this.defaultVisibleAttributes.result, (acc, next) => {
-
-                acc[next.clinicalAttributeId] = this.studyViewClinicalDataCountsCache
-                    .get({ attribute: next, filters: {} as any })
+                const clinicalDataType: ClinicalDataType = next.patientAttribute ? 'PATIENT' : 'SAMPLE';
+                const uniqueKey = clinicalDataType + '_' + next.clinicalAttributeId;
+                acc[uniqueKey] = this.studyViewClinicalDataCountsCache
+                    .get({ attribute: next, filters: this.emptyFilter })
                     .result!;
                 return acc;
             }, {} as any);
         },
         default: {},
-        onResult:() => {
-            this.initialized=true;
+        onResult: (result) => {
+            _.forEach(result, (obj, uniqueKey) => {
```

Comment:
Should we restart the conversation about the same uniqueKey(clinical attribute) in multiple studies?

Reply:
Sure, I think its better to deal with this logic in backend